src/uct/ib/rc/base/rc_iface.h

```diff
@@ -385,15 +393,19 @@ static UCS_F_ALWAYS_INLINE void
 uct_rc_iface_put_send_op(uct_rc_iface_send_op_t *op)
 {
     uct_rc_iface_t *iface = op->iface;
-    ucs_assert(op->flags == UCT_RC_IFACE_SEND_OP_FLAG_IFACE);
-    op->next = iface->tx.free_ops;
-    iface->tx.free_ops = op;
+
+    ucs_assert(op->flags & UCT_RC_IFACE_SEND_OP_FLAG_IFACE);
+
+    op->next                 = iface->tx.free_ops;
+    iface->tx.free_ops       = op;
+    iface->tx.ops_available += !!(op->flags & UCT_RC_IFACE_SEND_OP_FLAG_FLUSH);
```

Comment:
can we somehow reduce SW overhead here?

Reply:
it is used for zcopy and flushes only, not sure we can do much with performance here

Reply:
how about a different op->handler for flush operations? instead of UCT_RC_IFACE_SEND_OP_FLAG_FLUSH?

Reply:
In this case I'll have to set the handler every time I get the op from the buffer.

Reply:
or set/restore it for flush ops only, or have additional iface->tx.free_ops ptr for flush